write-fonts/src/tables/gvar.rs

```diff
@@ -242,13 +268,17 @@ impl GlyphVariations {
                 });
             *count += 1;
         }
-
-        // find the one that saves the most bytes
-        let (pts, _) = point_number_counts
-            .into_iter()
-            // no use sharing points if they only occur once
-            .filter(|(_, (_, count))| *count > 1)
-            .max_by_key(|(_, (size, count))| (*count - 1) * *size)?;
+        // find the one that saves the most bytes; if multiple are tied, pick the
+        // first one like python max() does (Rust's max_by_key() would pick the last),
+        // so that we match the behavior of fonttools
+        let (pts, _) = max_by_first_key(
+            point_number_counts
+                .clone()
+                .into_iter()
```

Comment:
as trivia, there's an interesting distinction between `clone().into_iter()` and `iter().cloned()`. The former clones the whole collection first and then iterates over the moved values, and the former iterates over borrowed values, cloning them one at a time, which is slightly more efficient.


There's probably an even more efficient option, here. Since we only want the max from the collection it would be strictly best to avoid cloning at all during the iteration, and only clone the single winner.


Ultimately this doesn't matter much, but I think it's worth building up that instinct around what the code we're writing is doing, and if it's the most efficient option.

Reply:
sorry the clone is a leftover when I was printing the damn thing :)

Reply:
thanks for spotting!